web-api/src/mind.rs

```diff
@@ -231,6 +315,124 @@ where
         .map_err(Into::into)
 }
 
+// function that assigns a score to a vector of documents based on the user's interests
+// score is equal to 1 if the document is interesting to the user, 0 otherwise
+fn score_documents(documents: &[&Document], user_interests: &[String]) -> Array1<f32> {
+    let scores = documents
+        .iter()
+        .map(|document| {
+            if document.is_interesting(user_interests) {
+                2.0
+            } else {
+                0.0
+            }
+        })
+        .collect::<Vec<f32>>();
+    Array1::from(scores)
```

Comment:
:art: everything which has a `From` impl automatically derives a `Into` impl as well, which can make some calls more ergonomically. from time to time it can have some type hint issues though, but here the functions signature should be enough for the compiler to infer correctly.
```suggestion
    documents
        .iter()
        .map(|document| {
            if document.is_interesting(user_interests) {
                2.0
            } else {
                0.0
            }
        })
        .collect::<Vec<f32>>()
        .into()
```

Reply:
Okay, I did not know about this feature, thanks for the suggestion!